tests/different-inputs.t

```diff
@@ -0,0 +1,41 @@
+Integration tests for nCoV pipeline
+
+  $ pushd "$TESTDIR"/.. > /dev/null
+
+Note that running these tests will pollute a number of directories
+(ncov/data, ncov/results, mcov/auspice, ncov/logs, ncov/benchmarks)
+as we don't have the ability to parameterise these.
+
+Before running each test we must clean out these directories, otherwise
+snakemake may use intermediate files from previous runs, thus producing
+inconsistent test results. This has the unfortunate side-effect of removing
+intermediate files from any runs which we may wish to keep.
```

Comment:
This approach seems fine for Travis CI runs, but I agree it's frustrating for local tests. There is a Snakemake command line argument `--directory` that allows you to set the working directory for the workflow. We could setup the `tests` directory to mirror the top-level directory (just the files it needs like `defaults/` and `data/`) and then run the workflow from the top-level directory with `snakemake --directory tests/ ...`. Would you want to try this out, @jameshadfield?

Reply:
I knew snakemake would have some magic for this. I'll check it out. I'm hesitant to copy & paste defaults etc, as that will lead to divergence and misleading test results, but we can probably make the cram set-up copy over the directory etc.

Reply:
Completely rewrote the tests to use this approach and all is working 💯 locally. I'm currently invoking cram with `--preserve-env` but there may be alternative / better approaches. 

Now running `cram --preserve-env tests/different-inputs.t` only modifies the (gitignored) `./tests/output` directory so it can be run independently of "normal" workflows.

Reply:
The tests are running locally, but are failing on GitHub CI when running `sanitize_metadata.py` on the (n=1) `references_metadata.tsv`. @huddlej have you seen this before? [Relevant logs](https://github.com/nextstrain/ncov/runs/2893322415?check_suite_focus=true):

```
python3 scripts/sanitize_metadata.py  --metadata data/references_metadata.tsv   ...
Traceback (most recent call last):
...
_csv.Error: Could not determine delimiter
```

Reply:
I'm going to take out the GitHub actions running of the tests and create a separate PR. They are failing in strange ways which are hard to debug.